server/src/main/java/org/elasticsearch/action/admin/indices/analyze/TransportAnalyzeAction.java

```diff
@@ -163,7 +163,7 @@ protected AnalyzeResponse shardOperation(AnalyzeRequest request, ShardId shardId
             }
             final AnalysisRegistry analysisRegistry = indicesService.getAnalysis();
             final int maxTokenCount = indexService == null ?
-                IndexSettings.MAX_TOKEN_COUNT_SETTING.get(settings) : indexService.getIndexSettings().getMaxTokenCount();
+                IndexSettings.MAX_TOKEN_COUNT_SETTING.get(Settings.EMPTY) : indexService.getIndexSettings().getMaxTokenCount();
```

Comment:
why Settings.EMPTY here?

Reply:
Ah! Good catch. I was going to leave a comment about it on the PR but forgot. I think I'll go with using `settings` here instead of this.

So you can specify `index.analyze.max_token_count` in elasticsearch.yml. Or on an index. I'm not sure why. But you can. When I made this change I thought "there is no way we document this so we don't need to support reading it from elasticsearch.yml" but I think that is not a good change to be sneaking into this.